source/grid/grid_out.cc

```diff
@@ -944,10 +944,17 @@ GridOut::write_dx(const Triangulation<dim, spacedim> &tria,
           << " data follows" << '\n';
       for (cell = tria.begin_active(); cell != endc; ++cell)
         {
-          // Little trick to get -1
-          // for the interior
+          // Little trick to get -1 for the interior
           for (unsigned int f = 0; f < GeometryInfo<dim>::faces_per_cell; ++f)
-            out << ' ' << (int)(signed char)cell->face(f)->boundary_id();
+            {
+              const types::boundary_id boundary_id =
+                cell->face(f)->boundary_id();
+              out << ' '
+                  << (boundary_id == static_cast<types::boundary_id>(-1) ?
```

Comment:
we have `numbers::invalid_material_id` that should probably be used on the right hand side.